gems/aws-sdk-core/lib/aws-sdk-core/binary/event_parser.rb

```diff
@@ -0,0 +1,92 @@
+module Aws
+  module Binary
+    # @api private
+    class EventParser
+
+      include Seahorse::Model::Shapes
+
+      # @param [Class] parser_class
+      # @param [Seahorse::Model::ShapeRef] rules
+      def initialize(parser_class, rules)
+        @parser_class = parser_class
+        @rules = rules
+      end
+
+      # Parse raw event message into event struct
+      # based on its ShapeRef
+      #
+      # @return [Struct] Event Struct
+      def apply(raw_event)
+        parse(raw_event)
+      end
+
+      private
+
+      def parse(raw_event)
+        message_type = raw_event.headers.delete(":message-type")
+        if message_type
+          case message_type.value
+          when 'error'
+            parse_error_event(raw_event)
+          when 'event'
+            parse_event(raw_event)
+          else
+            raise 'Unrecognized :message-type value for the event'
```

Comment:
Raise an appropriate typed error.

Reply:
Fixed, will be `EventStreamParserError.new ...`